app/modals/transaction/transaction-modal.tsx

```diff
@@ -148,22 +144,19 @@ export const TransactionModal: FC<TxModalProps> = ({ balance, address }) => {
     [publicKey]
   );
 
-  const signLedgerTransaction = useCallback(
-    async (unsignedTx: StacksTransaction) => {
-      if (!blockstackApp) throw new Error('``blockstackApp` is not defined');
-      const resp = await blockstackApp.sign(STX_DERIVATION_PATH, unsignedTx.serialize());
-      if (resp.returnCode !== LedgerError.NoErrors) {
-        throw new Error('Ledger responded with errors');
-      }
-      if (unsignedTx.auth.spendingCondition) {
-        (unsignedTx.auth.spendingCondition as any).signature = createMessageSignature(
-          resp.signatureVRS.toString('hex')
-        );
-      }
-      return unsignedTx;
-    },
-    [blockstackApp]
-  );
+  const signLedgerTransaction = useCallback(async (unsignedTx: StacksTransaction) => {
+    const serializedTx = unsignedTx.serialize().toString('hex');
+    const resp = await api.ledger.signTransaction(serializedTx);
+    if (resp.returnCode !== LedgerError.NoErrors) {
+      throw new Error('Ledger responded with errors');
+    }
+    if (unsignedTx.auth.spendingCondition) {
+      (unsignedTx.auth.spendingCondition as any).signature = createMessageSignature(
+        resp.signatureVRS
+      );
+    }
+    return unsignedTx;
+  }, []);
```

Comment:
```suggestion
  }, [api]);
```

Reply:
this `api` actually refers to a global object defined in the preload.js, rather than the Api api from the hook that does change. So it's not needed right?

Reply:
If it does not change, then we don't need to include it